android/java/org/chromium/chrome/browser/customtabs/FullScreenCustomTabActivity.java

```diff
@@ -116,6 +119,21 @@ public void performPostInflationStartup() {
                     .writeInt(
                             BravePermissionUtils.REWARDS_NOTIFICATION_PERMISSION_COUNT, count + 1);
         }
+
+        getOnBackPressedDispatcher()
```

Comment:
I think what we need here is to `Override` `finish()` and set the boolean there. I'm just worry that there are other cases that we can miss and not only back press, system closes the Activity etc

Reply:
@SergeyZhukovsky i am using onDestroy here to be 100% sure.